lib/nghttp2_session.c

```diff
@@ -3163,7 +3163,7 @@ ssize_t nghttp2_session_mem_send(nghttp2_session *session,
 }
 
 int nghttp2_session_send(nghttp2_session *session) {
-  const uint8_t *data;
+  const uint8_t *data = NULL;
```

Comment:
I think you're correct that it's not necessary (in fact, I'm pretty sure this entire thing is a gcc bug, which I'm trying to make a test case for), but if something changes in nghttp2_session_mem_send_internal() the initialization here will keep gcc happy. 

I only included it because I I didn't look to see if anything else calls nghttp2_session_mem_send_internal() and i'm not sure if there might be any reason to change that. Either way I'm happy.